flink-runtime/src/main/java/org/apache/flink/runtime/deployment/InputChannelDeploymentDescriptor.java

```diff
@@ -103,9 +103,7 @@ public String toString() {
 			// The producing task needs to be RUNNING or already FINISHED
 			if (consumedPartition.isConsumable() && producerSlot != null &&
 					(producerState == ExecutionState.RUNNING ||
-						producerState == ExecutionState.FINISHED ||
-						producerState == ExecutionState.SCHEDULED ||
-						producerState == ExecutionState.DEPLOYING)) {
```

Comment:
This change will break deployments where `allowLazyDeployment` is set to `false`. In the Flip-6 code, this is per default set to `true`, but for the legacy mode, the value is `false`. Thus, we would have to first remove the legacy mode and `allowLazyDeployment`.

Reply:
Thank you till. But where is legacy code ? I found a lot of code belongs to legacy mode.........omg....I do not think I should remove all legacy mode code....... Could you tell me please ? Lol.

Reply:
Yes, I would say that we tackle this problem after we have removed the legacy code. For the moment, there is a work around to set the max backoff time higher for the data connections.

Reply:
Yes Till. The flink1.5 code does not use JobManager and TaskManager class for running. Instead, we use JobMaster & TaskExecutor now. I think we should remove them and the relative class for legacy code.